datastore/src/main/scala/org/allenai/datastore/Datastore.scala

```diff
@@ -162,6 +80,16 @@ class Datastore(val name: String, val s3: AmazonS3Client) extends Logging {
     cause
   )
 
+  /** Exception indicating that the locator provided was for the wrong datastore. This is thrown if
+    * the "datastore" field in a locator doesn't match this datastore's name.
+    * @param locator the bad locator
+    */
+  class BadLocatorException(locator: Locator)
+    extends DsException(
+      s"locator for ${locator.datastore} doesn't match this datastore ($name)",
+      null
+    )
```

Comment:
You know, this is actually kind of a weird design now. The methods in the `Datastore` class could handle other buckets just fine. The only reason you have to create multiple `Datastore` instances is because it throws this exception otherwise.

Maybe it's cleaner to have only one `Datastore` object, and have all the shenanigans with generating the bucket name and the S3 keys in the locator, as methods?